cdap-api/src/main/java/co/cask/cdap/api/RunnableContext.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api;
+
+import co.cask.cdap.api.data.stream.StreamWriter;
+import co.cask.cdap.api.templates.AdapterContext;
+
+/**
+ * Defines an interface for Runnable components which need access to datasets, services, streams.
+ */
+public interface RunnableContext extends RuntimeContext, ServiceDiscoverer, StreamWriter, AdapterContext {
```

Comment:
How did you choose the name ``Runnable``? It has a history in CDAP, so wondering how :)

Either way, we may have to think of a better name.